File: lib/elements/text.ts
import { Element } from "../element.ts";
import { State } from "../state.ts";

export class TextElement extends Element {
    public _text = ""

    public set text(val: string) {
        this._text = val
        this.set(State.Text, val) 
    }

    public get text(): string {
        return this._text
    }
}

File: src/nodes/text.rs
use super::super::state::State;

#[derive(Clone)]
pub struct TextNode {
    pub text: String,
}

impl TextNode {
    pub fn new() -> Self {
        Self {
            text: String::from(""),
        }
    }

    pub fn set_state(&mut self, state: State) {
        match state {
            State::Text(text) => self.text = text,
            _ => panic!()
        };
    } 
}